internal/restore/volumesnapshot_restorer.go

```diff
@@ -58,16 +62,61 @@ func (p *VSRestorer) Execute(input *velero.RestoreItemActionExecuteInput) (*vele
 		return &velero.RestoreItemActionExecuteOutput{}, errors.Wrapf(err, "failed to convert input.Item from unstructured")
 	}
 
-	var vsFromBackup snapshotv1beta1api.VolumeSnapshot
-	if err := runtime.DefaultUnstructuredConverter.FromUnstructured((input.ItemFromBackup.UnstructuredContent()), &vsFromBackup); err != nil {
-		return &velero.RestoreItemActionExecuteOutput{}, errors.Wrapf(err, "failed to convert input.ItemFromBackup from unstructured")
+	labels := vs.Labels
```

Comment:
Can you add 1+ comments explaining the addition of L65-115? The method godoc should be updated, and any more detailed info can go inline.